### PR TITLE
Keycard initial integration

### PR DESCRIPTION
**Pull request by gravityblast**
Initial integration of Keycard using the nim/go wrapper https://github.com/status-im/nim-keycard-go.

The only command implemented in status-lib now is the `select` command. 
If everything works in desktop, we can start implementing here all the commands already implemented in the wrapper.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -111,37 +115,49 @@ export STATUSGO
...
+EXTRA_LIBS_DYNAMIC := -L"$(CURDIR)/build" -lstatuslib -lm -L"$(STATUSGO_LIBDIR)" -lstatus -L"$(KEYCARDGO_LIBDIR)" -lkeycard
 build_ctest: | $(LIBSTATUSLIB) build deps
 	echo -e $(BUILD_MSG) "build/ctest" && \
 		 $(CC) test/main.c -Wl,-rpath,'$$ORIGIN' -I./vendor/nimbus-build-system/vendor/Nim/lib $(EXTRA_LIBS_DYNAMIC) -g -o build/ctest
 
 LD_LIBRARY_PATH_NIMBLE := $${LD_LIBRARY_PATH}
 ifneq ($(detected_OS),Windows)
  ifneq ($(detected_OS),Darwin)
-  LD_LIBRARY_PATH_NIMBLE := $(STATUSGO_LIBDIR):$(LD_LIBRARY_PATH_NIMBLE)
+	LD_LIBRARY_PATH_NIMBLE := $(STATUSGO_LIBDIR):$(KEYCARDGO_LIBDIR):$(LD_LIBRARY_PATH_NIMBLE)
```
Comment (michaelsbradleyjr) [inline Makefile]: My understanding is that it's best to not use tabs in non-recipe sections of a Makefile, though that understanding could be outdated or I misunderstood the guidance re: when to use and not use tabs.

File: Makefile [hunk 2]
```diff
@@ -111,37 +115,49 @@ export STATUSGO
...
  endif
 endif
 
 PATH_NIMBLE := $${PATH}
 ifeq ($(detected_OS),Windows)
- PATH_NIMBLE := $(STATUSGO_LIBDIR):$(PATH_NIMBLE)
+	PATH_NIMBLE := $(STATUSGO_LIBDIR):$(KEYCARDGO_LIBDIR):$(PATH_NIMBLE)
```
Comment (michaelsbradleyjr) [inline Makefile]: See my previous comment re: tabs in non-recipe sections of a Makefile.

**Review by michaelsbradleyjr — DISMISSED** [review]
Tests workflow is failing in macOS because `install_name_tool` needs to be used on `libkeycard.dylib` like we do for `libstatus.dylib`:

https://github.com/status-im/status-lib/blob/keycard/nim_status_lib.nimble#L47-L49

Test workflow is failing in Linux due to, I think, a missing package:

https://github.com/status-im/status-lib/pull/63/checks?check_run_id=3756546833#step:8:86

```
# pkg-config --cflags  -- libpcsclite
Package libpcsclite was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcsclite.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libpcsclite' found
pkg-config: exit status 1
```

Maybe  it's something we can install at the start of the workflow with `apt`?

https://packages.ubuntu.com/focal/libpcsclite-dev

**Comment by michaelsbradleyjr** [comment]
I added several commits re: the changes I requested. The GHA tests workflow is now passing on all platforms.

**Comment by michaelsbradleyjr** [comment]
I've just rebased this PR branch against current `master`.

**Comment by gravityblast** [comment]
thank you @michaelsbradleyjr , I also added a new commit and then this PR should be ready to be reviewed/merged

File: status/types/keycard.nim [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import options
+
```
Comment (iurimatias) [inline status/types/keycard.nim]: types should be moved to the root ./types/keycard.nim

**Comment by gravityblast** [comment]
@iurimatias I moved the types and from the statusgo backend I catch the exception and re-raise it

**Pull request merged**